polus-multichannel-tiff-plugin/src/main.py

```diff
@@ -0,0 +1,158 @@
+from bfio.bfio import BioReader, BioWriter
+import bioformats
+import javabridge as jutil
+import argparse, logging, subprocess, multiprocessing, filepattern
+import numpy as np
+from pathlib import Path
+from concurrent.futures import ThreadPoolExecutor, wait
+from threading import Lock
+from multiprocessing import cpu_count
+import time
+
+# Thread lock for file writing
+lock = Lock()
+
+def write_tile(br,bw,X,Y,Z,C):
+    """Read and write one tile of a multichannel image
+    
+    This function is intended to be run inside of a thread. Since data is sent
+    to the JVM and the GIL is released, reading images can be performed in
+    parallel. Writing still requires a lock.
+
+    Args:
+        br ([bfio.BioReader]): BioReader to be read from
+        bw ([bfio.BioWriter]): BioWriter to be written to
+        X ([list]): Range of x-values to read/write
+        Y ([list]): Range of y-values to grab/write
+        C ([list]): Channel to write
+    """
+    
+    # Connect to the JVM
+    jutil.attach()
+    
+    # Read the image
+    image = br.read_image(X=X,Y=Y,Z=Z)
+    
+    # Lock the thread and write the image
+    with lock:
+        bw.write_image(image,X=[X[0]],Y=[Y[0]],Z=[Z[0]],C=C)
+    
+    # Detach from the JVM
+    jutil.detach()
+
+if __name__=="__main__":
+    # Initialize the logger
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    logger = logging.getLogger("main")
+    logger.setLevel(logging.INFO)
+
+    ''' Argument parsing '''
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='Create multichannel, ome-tif from an image collection.')
+    
+    # Input arguments
+    parser.add_argument('--filePattern', dest='filePattern', type=str,
+                        help='Filename pattern used to separate data', required=True)
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                        help='Input image collection to be processed by this plugin', required=True)
+    parser.add_argument('--channelOrder', dest='channelOrder', type=str,
+                        help='Input image collection to be processed by this plugin', required=True)
+    
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+    filePattern = args.filePattern
+    logger.info('filePattern = {}'.format(filePattern))
+    inpDir = args.inpDir
+    if (Path.is_dir(Path(args.inpDir).joinpath('images'))):
+        # switch to images folder if present
+        fpath = str(Path(args.inpDir).joinpath('images').absolute())
+    logger.info('inpDir = {}'.format(Path(inpDir).resolve()))
+    channelOrder = [int(c) for c in str(args.channelOrder).split(',')]
+    logger.info('channelOrder = {}'.format(channelOrder))
```

Comment:
Please add a comment in the readme which asserts that the channel order should only be separated by a comma.